component/core/src/main/java/org/exoplatform/social/core/search/impl/SearchServiceImpl.java

```diff
@@ -76,6 +80,9 @@ public void addConnector(SearchConnectorPlugin connectorPlugin) {
 
   @Override
   public Set<SearchConnector> getEnabledConnectors() {
+    UserACL userACL = CommonsUtils.getService(UserACL.class);
+    boolean isExternalUser = userACL.isUserInGroup(PLATFORM_EXTERNALS_GROUP);
+    connectors.stream().filter(connector -> connector.getName().equals("space") || connector.getName().equals("people")).forEach(connector -> connector.setEnabled(!isExternalUser));
     return Collections.unmodifiableSet(connectors.stream()
                                                  .filter(SearchConnector::isEnabled)
```

Comment:
```suggestion
                                                 .filter(c -> c.isEnabled && (
                                                    (c.getName().equals("people") || c.getName().equals("space")) && !isExternalUser)))
```
We filter connectors : it should be enable AND if connector name is people or space, user should not be external

Reply:
Updated!